lib/trivial.nix

```diff
@@ -467,6 +471,29 @@ rec {
     then v
     else k: v;
 
+  /*
+    Make an function with a "set pattern" delay its evaluation of the attribute
+    set in the argument.
```

Comment:
I can neither parse nor understand this sentence. Took me a while to understand, but here's my suggestion for a better description:

> Turns functions of the form `{ a, b }: f a b` (which evaluates the argument before evaluating the function body) into `args: f args.a args.b` (which evaluates the argument only when one of its attributes is needed). This is especially useful for fixed-point functions, which are only compatible with the latter form.

Reply:
I've rewritten the docs to be more explicit. I wish I could add a link to the definition of _set pattern_ in the Nix manual, but it's not linkable, and not structured as a definition. :(